src/api-reference/things/things.service.ts

```diff
@@ -127,4 +130,10 @@ export class ThingsService {
       throw new InvalidThingDescriptionException(errors);
     }
   }
+
+  private async checkDuplicateThingDescriptionId(urn: string, id: string | undefined): Promise<void> {
+    if (!id || id === urn) return;
+    const idExist = await this.thingDescriptionRepository.exist({ where: { urn: id } });
+    if (idExist) throw new DuplicateIdException(id as string);
+  }
```

Comment:
The function signature could become clearer by renaming the parameters `urn` and `id` to something like `currentId` and `newId`.

Reply:
Done!